### ISSUE TITLE
CMAKE_INSTALL_PREFIX for install process

### ISSUE THREAD

**Issue by guestieng** [microsoft/vscode-cmake-tools#1511]
With cmake (directly) it is possible to install a project with "temporarily" setting the install prefix, in the sense that CMAKE_INSTALL_PREFIX is not changed within the CmakeCache file. 
The corresponding command would look like this:
`cmake --install . --config <build type> --prefix <install prefix>`
Is there a way to do this with cmake-tools, i.e., I guess `cmake.install`?

**Comment by bobbrow** [comment]
Does it work if you add `"--prefix",  "<install prefix>"` to the `cmake.buildArgs` array?

**Comment by guestieng** [comment]
I tried
`"cmake.buildArgs": ["--prefix", "<install prefix>" ] `
in the workspace settings (do you mean it this way?). 
This does not work and yields the error
```
MSBUILD : error MSB1001: unknown switch
Switch: --prefix.vcxproj
MSBUILD : error MSB1009: Project file does not exis.
Switch: <install prefix>.vcxproj
```
Otherwise I tried
`"cmake.buildArgs": ["--prefix <install prefix>" ] `.
This yields the error
```
MSBUILD : error MSB1001: unknown switch
Switch: --prefix <install prefix>.vcxproj
```

**Comment by bobbrow** [comment]
Yes, I meant the first one.  I haven't used it in my own projects, but looking at the code I thought it might have been supported.  It seems that the `--prefix` might only be supported for the newer `--install` switch which is not currently used by this extension.  It still generates a command like `--build . --target install`

**Comment by guestieng** [comment]
Alright, can one expect to have that any time soon?
Ideally, having the following 2 features would be nice:
1. A way to do installation only, e.g.: 
`cmake --install . --config <build type> --prefix <install prefix>`
2. A combination of building and pure installation, i.e., combining consecutively
`cmake --build . --target install` or `cmake --build . `
and
 `cmake --install . --config <build type> --prefix <install prefix>`

**Comment by bobbrow** [comment]
I can't promise when this might be done, but we would accept a PR from the community that implements this behavior.  It's possible all that needs to happen is to update the command based on the version of cmake being used and expose a new setting `cmake.installArgs` that will allow you to set additional switches during an "install".

Number 1. should definitely be implemented.  Number 2. seems to be a lower priority to us since you would already be able to invoke both commands in sequence, but could perhaps be implemented by creating a new setting for it.

**Comment by gcampbell-msft** [comment]
@guestieng CMake Presets supports the `installDir` field which should help you solve this. We also have the `CMake: Install` command. 

Do these things help solve this issue?
